src/live_restore/live_restore_fs.c

```diff
@@ -0,0 +1,1248 @@
+                 * currently has a length of zero, but we want its length to be the same as the
+                 * source file.
+                 */
+                wt_off_t source_size;
+
+                lr_fh->source->fh_size(lr_fh->source, wt_session, &source_size);
+                __wt_verbose_debug1(session, WT_VERB_FILEOPS,
+                  "Creating destination file backed by source file. Copying size (%ld) from source "
+                  "file",
+                  source_size);
+
+                /*
+                 * Set size by truncating. This is a positive length truncate so it actually extends
+                 * the file. We're bypassing the live_restore layer so we don't try to modify the
+                 * extents in hole_list.
+                 */
+                lr_fh->destination.fh->fh_truncate(lr_fh->destination.fh, wt_session, source_size);
+
+                /*
+                 * Initialize the extent as one hole covering the entire file. We need to read
+                 * everything from source.
+                 */
+                WT_ERR(__live_restore_alloc_extent(
+                  session, 0, (size_t)source_size, NULL, &lr_fh->destination.hole_list));
+            }
+        } else
+            lr_fh->destination.complete = true;
+    }
+
+    /* Initialize the jump table. */
+    lr_fh->iface.close = __live_restore_fh_close;
+    lr_fh->iface.fh_lock = __live_restore_fh_lock;
+    lr_fh->iface.fh_read = __live_restore_fh_read;
+    lr_fh->iface.fh_size = __live_restore_fh_size;
+    lr_fh->iface.fh_sync = __live_restore_fh_sync;
+    lr_fh->iface.fh_truncate = __live_restore_fh_truncate;
+    lr_fh->iface.fh_write = __live_restore_fh_write;
+
+    /* TODO: These are unimplemented. */
+    lr_fh->iface.fh_advise = NULL;
+    lr_fh->iface.fh_sync_nowait = NULL;
+    lr_fh->iface.fh_unmap = NULL;
+    lr_fh->iface.fh_map_preload = NULL;
+    lr_fh->iface.fh_map_discard = NULL;
+    lr_fh->iface.fh_map = NULL;
+    lr_fh->iface.fh_extend = NULL;
+    lr_fh->iface.fh_extend_nolock = NULL;
+
+    *file_handlep = (WT_FILE_HANDLE *)lr_fh;
+
+    if (0) {
+err:
+        if (lr_fh != NULL)
+            __live_restore_fh_close((WT_FILE_HANDLE *)lr_fh, wt_session);
+    }
+    return (ret);
+}
+
+/*
+ * __live_restore_fs_remove --
+ *     Remove a file. We can only delete from the destination directory anyway.
+ */
+static int
+__live_restore_fs_remove(
+  WT_FILE_SYSTEM *fs, WT_SESSION *wt_session, const char *name, uint32_t flags)
+{
+    WT_DECL_RET;
+    WT_LIVE_RESTORE_FS *lr_fs;
+    WT_LIVE_RESTORE_FS_LAYER_TYPE layer;
+    WT_SESSION_IMPL *session;
+    char *path;
+    bool exist;
+
+    session = (WT_SESSION_IMPL *)wt_session;
+    lr_fs = (WT_LIVE_RESTORE_FS *)fs;
+
+    WT_UNUSED(layer);
+    exist = false;
+    path = NULL;
+
+    WT_RET_NOTFOUND_OK(__live_restore_fs_find_layer(fs, session, name, &layer, &exist));
+    if (ret == WT_NOTFOUND || !exist)
+        return (0);
```

Comment:
I think this should be `return (ENOENT);`? The posix layer calls `unlink(1)` which doesn't document what happens when there's no file to delete, but it points to `unlink(2)` which will throw an ENOENT